analyses/survival-analysis/util/quantile_calc.R

```diff
@@ -0,0 +1,52 @@
+# Functions for calculating quantiles of variables used for survival analysis
+
+# using 0.5 as a cutoff
+calc_groups <- function(meta_indep, tp53_quantiles, telomerase_quantiles){
+  meta_indep$tp53_0.5 <- ifelse(meta_indep$tp53_score > 0.5, "tp53 high",
+                                ifelse(meta_indep$tp53_score <= 0.5, "tp53 low",NA))
+  
+  # using 0.5 as a cutoff
+  meta_indep$tel_0.5 <- ifelse(meta_indep$telomerase_score > 0.5, "telomerase high",
+                               ifelse(meta_indep$telomerase_score <= 0.5, "telomerase low",NA))
```

Comment:
The nested `ifelse`s in this function are difficult to follow, but can be simplified and made easier to follow with `dplyr` and specifically `case_when`. I am providing a partial example here, but the same changes could be made throughout this function. I think that all of the added columns could be included in a single call to `mutate`

```suggestion
 meta_indep <- meta_indep %>%
  dplyr::mutate(
    tp53_0.5 = dplyr::case_when(
      tp53_score > 0.5  ~ "tp53 high",
      tp53_score <= 0.5 ~ "tp53 low"),
    tel_0.5 = dplyr::case_when(
      telomerase_score > 0.5  ~ "telomerase high",
      telomerase_score <= 0.5 ~ "telomerase low")
    )
```

Note that the `NA` case is handled automatically.

Reply:
@runjin326 can you work on this piece